### ISSUE TITLE
Add license file

### ISSUE THREAD

**Issue by Treehopper** [scream3r/java-simple-serial-connector#79]
The java files have this header:
> You should have received a copy of the GNU Lesser General Public License
> along with jSSC. [...]
But I cannot find such a file.
Currently I'd have to go through all source files to be sure about the license.

**Comment by mas2ao** [comment]
Treehopper
I'm not sure, but I think I got the license with the source code from here
https://code.google.com/p/java-simple-serial-connector/
on "Download"
correct if I'm wrong or if that license file is useless

**Comment by woodrowbarlow** [comment]
I agree that it would be a good idea to add a license file to the repo, but it's not strictly necessary. code comments say that it uses LGPL, which you can find in its full text here: https://www.gnu.org/licenses/lgpl.html.

You can assume version 3 since, even if it was released under version 1 or 2, GPL licenses are forwards-compatible.

This is unlikely to be added to this repo, but the maintainer is stating that they are working on a new release (largely a full re-write, from the sounds of it), so hopefully the next release will have a license file included in the repo.

**Comment by scream3r** [comment]
License files are packed with release binaries, and you can find it here https://github.com/scream3r/java-simple-serial-connector/releases

**Comment by Treehopper** [comment]
@woodrowbarlow From a legal point of view this might be OK. It's just inconvenient to have to check all file headers.

@scream3r Thanks for pointing this out. I didn't see it before. I still find it confusing though, because there are two license files. If I wouldn't have checked the file headers, I wouldn't know if maybe some files are under GPL and others are under LGPL, or if the GPL license is just there as a reference.
Maybe the README.txt should point that out?

I don't have a strong opinion on how that kind of thing should be handled, but I think it would make your project more "user-friendly". At least I was confused, so others might be too.
If you don't see any harm in that, I would recommend just adding a LGPLv3.txt with the license to the root of the repository.

**Comment by nicola-lunghi** [comment]
@scream3r 
Hi sorry to reopen this but  I'm working on a yocto recipe for this library.
Without a common LICENSE file basically legally is like every source file has a different license, 
so in the yocto recipe I have to specify the md5sum and the path of every source file to double check if the license has changed
like this

LIC_FILES_CHKSUM = " \
    file://${S}/src/cpp/_nix_based/jssc.cpp;endline=24;md5=aaaaaaaaaaaaaaaaaaaaaaa \
    file://${S}/src/cpp/jssc_SerialNativeInterface.h;endline=24;md5=aaaaaaaaaaaaaaaaaaaaaaa \
... (also all java files)
"
Could be very useful if you can add a global license file.
Or at least copy the license file also in the source release
Thanks
